### ISSUE TITLE
saving via joypad causes fire in-game

### ISSUE THREAD

**Issue by jmtd** [chocolate-doom/chocolate-doom#895]
Introduced by merging the work started in #893. I decided it was better to have the ability to save via joypad even though it introduces this bug (since there's a workaround: just reload after you save). Some good analysis as to the cause and outline of a solution by @Wintermute0110 in the comments for that issue.

**Comment by Wintermute0110** [comment]
@jmtd Thanks for the last merging. I have not tested it yet but I'm sure it will work fine.

I've thinking further about the joystick interface and have some ideas about the implementation. However, there are several decisions to made before starting the work. For example, if we want to properly support several joystick then a new configuration file needs to be created. Writing everything will take me some time and in addition I've seen that a lot of useful information and comments about the joystick interface are spread out in several issues.

Do you want to start the discussion about the new joystick implementation here or we keep talking on issue #758?

**Comment by jmtd** [comment]
To fix this bug, I wondered whether to add some stuff to the main event handler that lets us ignore joystick events for a few tics, basically like `joywait = I_GetTime() + 5` in the menu code but in the main loop. And we would only set this `joywait` for this situation (and any other such time things like this come up).

> Do you want to start the discussion about the new joystick implementation here or we keep talking on issue #758?

Maybe we should create a new issue as a "new joystick interface" talking point and keep this and #758 focussed on those specific sub-issues.

**Comment by Wintermute0110** [comment]
> Maybe we should create a new issue as a "new joystick interface" talking point and keep this and #758 focussed on those specific sub-issues.

Sounds reasonable.
